app/components/instruction/__tests__/AssociatedTokenDetailsCard.spec.tsx

```diff
@@ -0,0 +1,62 @@
+import { BaseInstructionCard } from '@components/common/BaseInstructionCard';
+import { intoTransactionInstructionFromVersionedMessage } from '@components/inspector/utils';
+import * as spl from '@solana/spl-token';
+import { PublicKey } from '@solana/web3.js';
+import { render, screen } from '@testing-library/react';
+
+import * as stubs from '@/app/__tests__/mock-stubs';
+import * as mock from '@/app/__tests__/mocks';
+import { ClusterProvider } from '@/app/providers/cluster';
+import { ScrollAnchorProvider } from '@/app/providers/scroll-anchor';
+
+import { ParsedInstructionFactory } from '../../inspector/utils';
+import { AssociatedTokenDetailsCard } from '../associated-token/AssociatedTokenDetailsCard';
+
+jest.mock('next/navigation');
+
+describe('instruction::AssociatedTokenDetailsCard', () => {
+    beforeEach(() => {
+        mock.mockUseSearchParams();
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    const factory = ParsedInstructionFactory();
+
+    test('should render "CreateIdempotentDetailsCard"', async () => {
+        const index = 1;
+        const m = mock.deserializeMessageV0(stubs.aTokenMsg);
+        const ti = intoTransactionInstructionFromVersionedMessage(m.compiledInstructions[index], m);
+        expect(ti.programId.equals(spl.ASSOCIATED_TOKEN_PROGRAM_ID)).toBeTruthy();
+
+        const parsed = {
+            account: new PublicKey('Fv8YYjF2DUqj9RZhyXNzXa4yR9nHHwjg5bFjA82UidF1'),
+            mint: new PublicKey('74SBV4zDXxTRgv1pEMoECskKBkZHc2yGPnc7GYVepump'),
+            source: new PublicKey('EzdQH5zUfTMGb3vwU4oumxjVcxKMDpJ6dB78pbjfHmmb'),
+            systemProgram: new PublicKey('11111111111111111111111111111111'),
+            tokenProgram: new PublicKey('TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA'),
+            wallet: new PublicKey('EzdQH5zUfTMGb3vwU4oumxjVcxKMDpJ6dB78pbjfHmmb'),
+        };
+
+        const ix = factory.intoParsedInstruction(ti, parsed);
+        const tx = factory.intoParsedTransaction(ti, m);
```

Comment:
why isn't this just done with a normal function? Why is factory pattern needed here?

Reply:
I've answered at the previous comment